src/foam/nanos/crunch/ExpireUserCapabilityJunctionsCron.java

```diff
@@ -21,25 +21,23 @@
 
 import static foam.mlang.MLang.*;
 
+/**
+   Find all GRANTED junctions that are past the expiration date,
+   and if not in Grace period, expire them.
+ */
 public class ExpireUserCapabilityJunctionsCron implements ContextAgent {
 
-  private Logger logger;
-  private DAO userCapabilityJunctionDAO;
-
   @Override
   public void execute(X x) {
-    logger = Loggers.logger(x, this);
     DAO bareUserCapabilityJunctionDAO = (DAO) x.get("bareUserCapabilityJunctionDAO");
     DAO userCapabilityJunctionDAO = (DAO) x.get("userCapabilityJunctionDAO");
 
-    // Find all GRANTED junctions that are past the expiration date and if not in Grace period, expire them.
     List<UserCapabilityJunction> activeJunctions = ((ArraySink) userCapabilityJunctionDAO
       .where(
         AND(
           EQ(UserCapabilityJunction.STATUS, CapabilityJunctionStatus.GRANTED),
           NEQ(UserCapabilityJunction.EXPIRY, null),
-          GT(UserCapabilityJunction.GRACE_PERIOD, 0),
-          LT(UserCapabilityJunction.EXPIRY, new Date())
+          LTE(UserCapabilityJunction.EXPIRY, new Date())
```

Comment:
does this also need to check LTE(UserCapabilityJunction.GRACE_PERIOD, 0) ? 
when a ucj status is set to EXPIRED, and we are checking if the user 'has the capability', is there something that checks if the ucj is granted but also if it's expired but in graceperiod?

Reply:
[Yes, line 47 below.](https://github.com/kgrgreer/foam3/pull/2914/files/e6f3eccf1e30d60bf1fa1679e20caa571af9c29e..59740411f1cd78041700e70e808eeca47c29a592#diff-3ad866c5d79ddcfe979d2cdfe772b03bc9a9d5b0c9ce9637b480452d1cfc5effR47)